drivers/Kconfig

```diff
@@ -147,4 +147,6 @@ source "drivers/coredump/Kconfig"
 
 source "drivers/xen/Kconfig"
 
+source "drivers/ipc/Kconfig"
```

Comment:
We already have an IPC subsystem definitely we do not need a new IPC driver class for this.

Reply:
agree, why add a lone Kconfig in here? any folders under drivers/ is considered a subsystem that needs a defined API and drivers will need to be part of this subsystem.

Reply:
Ok, we can just check whether its enabled/disabled using single generic CONFIG_INTEL_ADSP_IPC and differentiate implementation by board/family definitions. Yes this can be removed thanks.
However I will need to leave existing CAVS_IPC symbol selecting INTEL_ADSP_IPC symbol just for the sake of CIs building SOF code that needs that symbol.